pyirf/irf/energy_dispersion.py

```diff
@@ -73,3 +75,86 @@ def energy_dispersion(
     energy_dispersion = _normalize_hist(energy_dispersion)
 
     return energy_dispersion
+
+
+def energy_dispersion_to_migration(
+    dispersion_matrix,
+    disp_true_energy_edges,
+    disp_migration_edges,
+    new_true_energy_edges,
+    new_reco_energy_edges,
+):
+    """
+    Construct a sparse energy migration matrix from a dense energy
+    dispersion matrix.
+    Depending on the new energy ranges, the sum over the first axis
+    can be smaller than 1.
+    The new true energy bins need to be a subset of the old range,
+    extrapolation is not supported.
+    New reconstruction bins outside of the old migration range are filled with
+    zeros.
+
+    Parameters
+    ----------
+    dispersion_matrix: numpy.ndarray
+        Energy dispersion_matrix
+    disp_true_energy_edges: astropy.units.Quantity[energy]
+        True energy edges matching the first dimension of the dispersion matrix
+    disp_migration_edges: numpy.ndarray
+        Migration edges matching the second dimension of the dispersion matrix
+    new_true_energy_edges: astropy.units.Quantity[energy]
+        True energy edges matching the first dimension of the output
+    new_reco_energy_edges: astropy.units.Quantity[energy]
+        Reco energy edges matching the second dimension of the output
+
+    Returns:
+    --------
+    migration_matrix: numpy.ndarray
+        Three-dimensional energy migration matrix. The third dimension
+        equals the fov offset dimension of the energy dispersion matrix.
+    """
+
+    migration_matrix = np.zeros((
+        len(new_true_energy_edges)-1,
+        len(new_reco_energy_edges)-1,
+        dispersion_matrix.shape[2],
+    ))
+
+    true_energy_interpolation = interpolate.interp1d(
+        (disp_true_energy_edges[1:] + disp_true_energy_edges[:-1])/2,
```

Comment:
Usually energy bins are logarithmically spaced, so maybe take the geometric mean here? If not just ignore my comment...